tensorflow/compiler/mlir/tosa/transforms/legalize_tfl.cc

```diff
@@ -2998,6 +2999,138 @@ LogicalResult ConvertTFLCosOp::matchAndRewrite(
   return success();
 }
 
+LogicalResult ConvertTFLAtan2Op::matchAndRewrite(
+    Operation* op, PatternRewriter& rewriter) const {
+  auto tfl_atan2_op = cast<TFL::Atan2Op>(op);
+  Location loc = op->getLoc();
+  Value input_y = tfl_atan2_op.getY();
+  RankedTensorType input_y_ty = input_y.getType().dyn_cast<RankedTensorType>();
```

Comment:
With the new casting mechanisms in LLVM one can now finally do

auto input_y_ty = dyn_cast<RankedTensorType>(input_y.getType());

prefer to not repeat type for these (for those that are outputs of cast's we use auto as type is clear locally).